extensions/default/src/DicomWebDataSource/utils/clientManager.ts

```diff
@@ -0,0 +1,249 @@
+import { api } from 'dicomweb-client';
+import StaticWadoClient from './StaticWadoClient';
+import dcm4cheeReject from '../dcm4cheeReject';
+import { errorHandler, utils } from '@ohif/core';
+
+/**
+ * This object plays the central role in OHIF's multiple server handling ability.
+ * It stores all servers configurations and handles all request headers generations
+ */
+interface ConfigToAdd {
+  url: string;
+  qidoRoot: string;
+  wadoRoot: string;
+  staticWado: boolean;
+  singlepart: boolean;
+  name: string;
+  qidoConfig: {
+    url: string;
+    staticWado: boolean;
+    singlepart: boolean;
+    headers: object;
+  };
+  wadoConfig: {
+    url: string;
+    staticWado: boolean;
+    singlepart: boolean;
+    headers: object;
+  };
+  qidoDicomWebClient: api.DICOMwebClient | StaticWadoClient;
+  wadoDicomWebClient: api.DICOMwebClient | StaticWadoClient;
+}
+export default class ClientManager {
+  clients;
+  userAuthenticationService;
+
+  constructor({ params, query, dicomWebConfig, userAuthenticationService }) {
+    this.clients = [];
+    this.userAuthenticationService = userAuthenticationService;
+    const configArray = Array.isArray(dicomWebConfig) ? dicomWebConfig : [dicomWebConfig];
```

Comment:
dicomWebConfigs? since it can be an array?

Reply:
we should be able to just create an instance of this class and later add the dicomWebConfig. So check for 

```
if (!dicomWebConfigurations){
  return
}
```

Reply:
I cant change that because i will need to change in the other clietns (local, json)